csharp/src/Apache.Arrow/Arrays/Date32Array.cs

```diff
@@ -64,7 +64,7 @@ public Date32Array(ArrayData data)
                 return default;
             }
 
-            return DateTimeOffset.FromUnixTimeMilliseconds(value.Value * MillisecondsPerDay);
+            return DateTimeOffset.FromUnixTimeMilliseconds((long)value.Value * MillisecondsPerDay);
```

Comment:
I wonder if it would be better to define `MillisecondsPerDay` as 

```C#
private const long MillisecondsPerDay = 86_400_000;
```

That way if we use it for anymore arithmetic/multiplication, it would automatically be long arithmetic. Thoughts?

Reply:
I thought of that - but then I think you have to cast away the long on the division side - albeit the long/long to int conversion overflow during division becomes more obvious that the (int * int) overflow in multiplication....  so it probably is best ti define the constant as long